src/yolo/nms.jl

```diff
@@ -84,41 +104,47 @@ batchout rows:
 - second-to-last row (end-1) has the class index.
 - The last row is the batch index
 """
-function perform_detection_nms(
-    batchout,
-    overlapThresh,
-    batchsize::Int
-)
-    output = []  # array of matrices
-
-    @views for b in 1:batchsize
-        # Get columns that belong to batch b
-        b_idxs = findall(x -> x == b, batchout[end, :])
-        if isempty(b_idxs)
+function perform_detection_nms(batchout, overlapThresh, batchsize::Int)
+    output = similar(batchout)
+    i = 1  # index for writing into `output`
+
+    for b in 1:batchsize
+        b_cols = findall(==(b), @view(batchout[end, :]))
+        if isempty(b_cols)
             continue
         end
-        page = batchout[:, b_idxs]
 
-        # For each class present in this batch
-        present_classes = unique(page[end-1, :])
-        for c in present_classes
-            # Gather all detections that match class c
-            c_idxs = findall(x -> x == c, page[end-1, :])
+        page = @view batchout[:, b_cols]
+        class_ids = @view page[end-1, :]
+
+        seen_classes = Set{eltype(class_ids)}()
+
+        @inbounds for (local_col_idx, cls) in enumerate(class_ids)
```

Comment:
I'm pretty sure this `@inbounds` applies to the entire body here. Not sure that is what you meant; or if it is, then you shouldn't need the nested one.